op-service/signer/transaction_args.go

```diff
@@ -34,33 +34,46 @@ type TransactionArgs struct {
 	Input *hexutil.Bytes `json:"input"`
 
 	AccessList *types.AccessList `json:"accessList,omitempty"`
-	ChainID    *hexutil.Big      `json:"chainId,omitempty"`
+	ChainID    *hexutil.U256     `json:"chainId,omitempty"`
 
 	// Custom extension for EIP-4844 support
 	BlobVersionedHashes []common.Hash `json:"blobVersionedHashes,omitempty"`
-	BlobFeeCap          *hexutil.Big  `json:"maxFeePerBlobGas,omitempty"`
+	BlobFeeCap          *hexutil.U256 `json:"maxFeePerBlobGas,omitempty"`
+
+	// Custom extension for EIP-7702 support
+	AuthList []types.SetCodeAuthorization `json:"authorizationList,omitempty"`
 }
 
-// NewTransactionArgsFromTransaction creates a TransactionArgs struct from an EIP-1559 or EIP-4844 transaction
+// NewTransactionArgsFromTransaction creates a TransactionArgs struct from an EIP-1559, EIP-4844, or EIP-7702 transaction
 func NewTransactionArgsFromTransaction(chainId *big.Int, from *common.Address, tx *types.Transaction) *TransactionArgs {
 	data := hexutil.Bytes(tx.Data())
 	nonce := hexutil.Uint64(tx.Nonce())
 	gas := hexutil.Uint64(tx.Gas())
+
+	// Ignore overflow errors. We are assuming core.Transaction does not have out-of-range values
+	value, _ := uint256.FromBig(tx.Value())
+	chainIdU256, _ := uint256.FromBig(chainId)
+	gasFeeCap, _ := uint256.FromBig(tx.GasFeeCap())
+	gasTipCap, _ := uint256.FromBig(tx.GasTipCap())
+	blobGasFeeCap, _ := uint256.FromBig(tx.BlobGasFeeCap())
```

Comment:
Nit: how about using a helper function `MustFromBig` which panics if there is an overflow?

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/14489/commits/2de13e60daecfcc067d58461f2052b8dcae4b101